src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -0,0 +1,49 @@
+package org.galatea.starter.service;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.galatea.starter.domain.IexLastTradedPrice;
+import org.galatea.starter.domain.IexSymbol;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * A layer for transformation, aggregation, and business required when retrieving data from IEX.
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class IexService {
+
+  @NonNull
+  private IexClient iexClient;
+
+
+  /**
+   * Get all stock symbols from IEX.
+   *
+   * @return a list of all Stock Symbols from IEX.
+   */
+  public List<IexSymbol> getAllSymbols() {
+    return iexClient.getAllSymbols();
+  }
+
+  /**
+   * Get the last traded price for each Symbol that is passed in.
+   *
+   * @param symbols the list of symbols to get a last traded price for.
+   * @return a list of last traded price objects for each Symbol that is passed in.
+   */
+  public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final List<String> symbols) {
+    if (CollectionUtils.isEmpty(symbols)) {
```

Comment:
I believe it's better to return Collections.emptyList() in this scenario rather than newArrayList(), although this is a minor thing - https://www.baeldung.com/java-collections-emptylist-new-list.

Also, a personal preference thing, but I prefer the following pattern in this scenario (less indentation and fewer brackets):

if (CollectionUtils.isEmpty(symbols))
    return Collections.emptyList();

return iexClient.getLastTradedPriceForSymbols(symbols.toArray(new String[0]));

Reply:
I agree with using `Collections.emptyList()` for a list that's intentionally empty, and I personally also tend towards omitting else blocks when the if returns something, but the if statement should still be formatted as a standard if statement - i.e. with brackets and indentation. Aside from the fact that checkstyle requires them, omitting brackets from single-line blocks can easily lead to errors if someone adds a line but neglects to add brackets.